File: src/main/java/io/bindernews/thegrackle/power/FireWithinPower.java
package io.bindernews.thegrackle.power;

import com.megacrit.cardcrawl.actions.common.ApplyPowerAction;
import com.megacrit.cardcrawl.core.AbstractCreature;
import com.megacrit.cardcrawl.powers.AbstractPower;
import io.bindernews.thegrackle.GrackleMod;

public class FireWithinPower extends BasePower {
    public static final String POWER_ID = GrackleMod.makeId(FireWithinPower.class);

    public FireWithinPower(AbstractCreature owner, int amount) {
        super(POWER_ID);
        setOwnerAmount(owner, amount);
    }

    @Override
    public void atStartOfTurn() {
        addToBot(new ApplyPowerAction(owner, owner, new FireheartPower(owner, amount), amount, true));
    }

    @Override
    public void updateDescription() {
        description = formatDesc(0, amount);
    }

    @Override
    public AbstractPower makeCopy() {
        return new FireWithinPower(owner, amount);
    }
}

File: src/main/kotlin/net/bindernews/grackle/power/FireWithinPower.kt
package net.bindernews.grackle.power

import com.megacrit.cardcrawl.actions.common.ApplyPowerAction
import com.megacrit.cardcrawl.core.AbstractCreature
import net.bindernews.grackle.GrackleMod

class FireWithinPower(owner: AbstractCreature, amount: Int) : BasePower(POWER_ID) {
    init {
        setOwnerAmount(owner, amount)
        loadRegion("devotion")
    }

    override fun atStartOfTurn() {
        addToBot(ApplyPowerAction(owner, owner, FireheartPower(owner, amount), amount, true))
    }

    override fun updateDescription() {
        description = formatDesc(0, amount)
    }

    companion object {
        @JvmField val POWER_ID = GrackleMod.makeId(FireWithinPower::class.java)
    }
}